src/Diagnostics.RuntimeHost/Controllers/DiagnosticControllerBase.cs

```diff
@@ -152,10 +152,10 @@ protected async Task<IActionResult> ExecuteQuery<TPostBodyResource>(TResource re
                             if (!VerifyEntity(invoker, ref queryRes)) return Ok(queryRes);
                             OperationContext<TResource> cxt = PrepareContext(resource, startTimeUtc, endTimeUtc);
 
-                            var responseInput = new Response() { Metadata = RemovePIIFromDefinition(invoker.EntryPointDefinitionAttribute, cxt.IsInternalCall) };
+                            var responseInput = new Response() { Metadata = RemovePIIFromDefinition(invoker.EntryPointDefinitionAttribute, cxt.IsInternalClient) };
                             invocationResponse = (Response)await invoker.Invoke(new object[] { dataProviders, cxt, responseInput });
                             invocationResponse.UpdateDetectorStatusFromInsights();
-                            isInternalCall = cxt.IsInternalCall;
+                            isInternalCall = cxt.IsInternalClient;
```

Comment:
I guess I dont uderstand what is InternalClient and how is it different from InternalCall?

Reply:
InternalCall determines whether the detector runs as if it is being called by internal or external. I didn't change this because it is already being used and I didn't want to break anyone. 

InternalClient means that it is being called from Applens, or any other internal source in the future. 

Reason is that lets say you want to run a detector in applens, but see the output as a customer would. You want to run a detector that is actually internalOnly. If you try to run that detector before with InternalCall = false, you would get a 404 because it would get filtered out. That is why you need the logic of internal vs external client and internal vs. external view separate.

Reply:
You can have an internal client that wants an internal view or an internal client that wants an external (customer) view. If it is an external client, then internal call will always be false.

Reply:
This seems too much confusing :) 

Instead of this, can we add a field like user-agent or caller (of string type or enum type
) which can be populated by clients and gets populated in detectors context.

Applens calls ==> 'applens'
ASC calls ==> 'asc'
Any other ==> 'other'

Reply:
or maybe rename the field ? bool type is easy in the scenario you described. However, I wonder if folks starts using InternalClient flag to make a decision for internal call vs external call.